src/software/util/make_enum_macro/make_enum_macro.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace
+{
+    /**
+     * Checks if a string representing enum values is valid.
+     *
+     * An enum string is valid if it does not contain any '=', meaning
+     * the enum is not being assigned any values manually
+     *
+     * @param str A string containing comma-separated enum value names
+     *
+     * @return True if the string is valid, and false otherwise
+     */
+    constexpr bool isEnumArgsValid(const std::string_view& str)
+    {
+        size_t index = str.find('=');
+        return index == std::string_view::npos;
+    }
+
+    /**
+     * Separates a string of comma-separated values into a vector
+     *
+     * @param enum_string A string of comma-separated enum values
+     *
+     * @return A vector of strings, where each string is a value from the original string
+     */
+    std::vector<std::string> separateEnumStrings(const std::string& enum_string)
+    {
+        std::vector<std::string> strings;
+        std::ostringstream temp;
+        for (size_t i = 0; i < enum_string.length(); i++)
+        {
+            if (isspace(enum_string[i]))
+            {
+                continue;
+            }
+            else if (enum_string[i] == ',')
+            {
+                strings.push_back(temp.str());
+                temp.str(std::string());
+            }
+            else
+                temp << enum_string[i];
```

Comment:
code style nit: add braces around this line

Reply:
:scream: fixed